fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/ValueIndexExpansionVisitor.java

```diff
@@ -48,7 +51,7 @@
  * Class to expand value index access into a candidate graph. The visitation methods are left unchanged from the super
  * class {@link KeyExpressionExpansionVisitor}, this class merely provides a specific {@link #expand} method.
  */
-public class ValueIndexExpansionVisitor extends KeyExpressionExpansionVisitor implements ExpansionVisitor<KeyExpressionExpansionVisitor.VisitorState> {
+public class ValueIndexExpansionVisitor extends KeyExpressionExpansionVisitor implements ExpansionVisitor<KeyExpressionExpansionVisitor.VisitorState>, IndexPredicateExpansion {
```

Comment:
I don't care really one way or another, but if the methods in  `IndexPredicateExpansion` are all _non_selfish, why not make the interface a class with a bunch of statics and just call them from here. The original comment was just about the extraction of the general code that can be applied in the context of any kind of index versus the code that can just be applied for value indexes. TLDNR, we don't have to use a trait mix-in approach. Up to you. Of course all of this is moot if any of the methods are selfish.

Reply:
That's a good point, thanks. I don't have a strong opinion one way of the other, but I like the _non_selfish analogy. So, I will just refactor it into a class with a bunch of static methods instead.